src/components/UserProfile/FoodTicket.tsx

```diff
@@ -0,0 +1,171 @@
+"use client";
+
+import { generateClient } from "aws-amplify/api";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { QRCode } from "react-qrcode-logo";
+
+import { type Schema } from "@/amplify/data/resource";
+import { createMessageAndCode } from "@/amplify/function/BusinessLogic/utils/crytography";
+import { getUpcomingFoodEventDetails } from "@/app/get-food-ticket/actions";
+import { useUser } from "@/components/contexts/UserContext";
+import { useQuery } from "@tanstack/react-query";
+
+import ProfileLinks from "./ProfileLinks";
+
+export default function UserFoodTicket() {
+  const client = generateClient<Schema>();
+
+  const userId = useUser().currentUser.userSub as string;
+
+  const { data, isFetching } = useQuery({
+    initialDataUpdatedAt: 0,
+    queryKey: ["User", userId],
+    queryFn: async () => {
+      const {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+      } = await getUpcomingFoodEventDetails(userId);
+
+      let verificationCode = "";
+
+      const { data, errors } = await client.queries.GetUserMessageCode({
+        userMessage: userId,
+      });
+      if (errors) {
+        console.log(errors);
+        verificationCode = "Error when searching for your message code";
+        return;
+      }
+
+      if (data) {
+        const json = JSON.parse(data.body as string);
+        const code = json["value"];
+
+        const createdVerificationCode = createMessageAndCode(userId, code);
+        verificationCode = createdVerificationCode;
+      } else {
+        verificationCode = "Error when searching for your message code";
+      }
+
+      return {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+        verificationCode,
+      };
+    },
+  });
+
+  // Not sure if we actually plan on showing their use code and i have no idea what its used for so for container purposes i will shorten it
+  const shortenString = (str: string, maxLen: number) => {
+    if (str) {
```

Comment:
Maybe ideen can clear this up but I have no idea what the usercode does and it was messing with the borders of the container so i made this just to shorten it

Reply:
@AllanKoder can verify but I'm pretty sure that's what your supposed to put as input to the QR code.

Reply:
Well if its the input to the QR code im assuming itd be appended to some sort of endpoint. Allan showed it in his preview so I just assumed that I should as well

I am assuming we will remove it later because its not something we have to show to the user

Reply:
```suggestion
    <QRCode
                  value="https://www.google.ca"
                  ...
```

I just need to know what to change this to